### PR TITLE
Added is-functions and CardTypeProfileRequest

### PR DESCRIPTION
**Pull request by kaffe2affe**

### CODE REVIEW

File: model/OrganisationUpdateRequest.ts [hunk 1]
```diff
@@ -5,3 +5,14 @@ export interface OrganisationUpdateRequest {
 	status: "ACTIVE" | "DELETED"
 	organisationLimitRequest?: OrganisationBalanceLimitResponse
 }
+export namespace OrganisationUpdateRequest {
+	export function is(value: OrganisationUpdateRequest | any): value is OrganisationUpdateRequest {
+		return (
+			typeof value == "object" &&
+			typeof value.mcc == "string" &&
```
Comment (ChriJ) [inline model/OrganisationUpdateRequest.ts]: mcc isn't part of the interface

File: model/OrganisationCreateRequest.ts [hunk 2]
```diff
@@ -3,3 +3,9 @@ import { OrganisationUpdateRequest } from "./OrganisationUpdateRequest"
 export interface OrganisationCreateRequest extends OrganisationUpdateRequest {
 	code: string
 }
+
+export namespace OrganisationCreateRequest {
+	export function is(value: OrganisationCreateRequest | any): value is OrganisationCreateRequest {
+		return typeof value == "object" && typeof value.code == "string" && OrganisationUpdateRequest.is(value)
```
Comment (ChriJ) [inline model/OrganisationCreateRequest.ts]: before i had a problem regarding the type system when using an is-function directly on value (maybe it works here), but we can do .is({...value}) to be safe

File: model/OrganisationConfig.ts [hunk 3]
```diff
@@ -22,3 +22,16 @@ export interface OrganisationConfig {
...
+
+export namespace OrganisationConfig {
+	export function is(value: OrganisationConfig | any): value is OrganisationConfig {
+		return (
+			value == undefined ||
+			(typeof value == "object" &&
+				(value.defaultModulrUsage == undefined ||
+					value.defaultModulrUsage == "SINGLE_USE" ||
+					value.defaultModulrUsage == "SINGLE_USE_ALLOW_TEST_AUTH" ||
+					value.defaultModulrUsage == "MULTIPLE_USE"))
```
Comment (ChriJ) [inline model/OrganisationConfig.ts]: looks like this needs some more things moving forward, maybe this is fine with how it's about to be used for the time being

Comment (ChriJ) [reply model/OrganisationConfig.ts]: also those values look like CardUsage (which maybe has an is-function)

File: model/OmnisetupRequest.ts [hunk 4]
```diff
@@ -14,3 +14,20 @@ export interface OmnisetupRequest {
...
+			typeof value == "object" &&
+			(value.flags == undefined || OmnisetupFlags.is(value.flags)) &&
+			OrganisationCreateRequest.is(value.organisation) &&
+			Array.isArray(value.providers) &&
+			value.providers.every((item: any) => OmnisetupProviderRequest.is(item)) &&
+			(value.organisationConfig == undefined || OrganisationConfig.is(value.organisationConfig)) &&
+			(value.internalOrganisationConfig == undefined ||
+				InternalOrganisationConfig.is(value.internalOrganisationConfig)) &&
+			(value.sharedRolesets == undefined ||
+				(Array.isArray(value.sharedRolesets) && value.sharedRolesets.every((item: any) => typeof item == "string")))
```
Comment (ChriJ) [inline model/OmnisetupRequest.ts]: users missing in this

File: model/OmnisetupProviderRequest.ts [hunk 5]
```diff
@@ -13,3 +13,17 @@ export interface OmnisetupProviderRequest {
...
+export namespace OmnisetupProviderRequest {
+	export function is(value: OmnisetupProviderRequest | any): value is OmnisetupProviderRequest {
+		return (
+			typeof value == "object" &&
+			ProviderCode.is(value.providerCode) &&
+			(value.credentials == undefined || CredentialRequest.is(value.credentials)) &&
+			(value.providerSetupCredentials == undefined || CredentialRequest.is(value.providerSetupCredentials)) &&
+			(value.suppliers == undefined || SupplierRequest.is(value.suppliers)) &&
+			CardTypeSpecification.is(value.cardType) &&
+			(value.useAs == undefined || typeof value.useAs == "string")
```
Comment (ChriJ) [inline model/OmnisetupProviderRequest.ts]: fundingAccounts missing

File: Client/Cards/index.ts [hunk 6]
```diff
@@ -111,7 +111,15 @@ export class Cards extends List<model.CardResponseV2 | model.CardResponse> {
 		)
 		return this.extractResponse<model.CardTypeResponseV2>(response)
 	}
-	//Possibly should be moved to its own class
+
+	async setCardTypesV2(cardTypeProfileRequest: model.CardTypeProfileRequest) {
+		const response = await this.connection.post<model.CardTypeProfileResponse>(
+			"v2/card/types/profiles",
```
Comment (ChriJ) [inline Client/Cards/index.ts]: this path + POST creates a new profile, and the name of the method leads me to believe it's not what was intended.

File: model/InternalOrganisationConfig.ts [hunk 7]
```diff
@@ -11,3 +11,13 @@ export interface InternalOrganisationConfig {
 	statementAppned?: boolean
 	statementRebuild?: boolean
 }
+
+export namespace InternalOrganisationConfig {
+	export function is(value: InternalOrganisationConfig | any): value is InternalOrganisationConfig {
+		return (
+			typeof value == "object" &&
+			(value.flags == undefined ||
+				(Array.isArray(value.flags) && value.flags.every((item: any) => OrganisationFlag.is(item))))
```
Comment (ChriJ) [inline model/InternalOrganisationConfig.ts:20]: probably more to this as well

File: Client/Cards/index.ts (hunk 6 above)
Comment (kaffe2affe) [reply Client/Cards/index.ts]: Changed name to createCardTypesV2

File: Client/Cards/index.ts [hunk 8]
```diff
@@ -112,7 +112,7 @@ export class Cards extends List<model.CardResponseV2 | model.CardResponse> {
 		return this.extractResponse<model.CardTypeResponseV2>(response)
 	}
 
-	async setCardTypesV2(cardTypeProfileRequest: model.CardTypeProfileRequest) {
+	async createCardTypesV2(cardTypeProfileRequest: model.CardTypeProfileRequest) {
```
Comment (ChriJ) [inline Client/Cards/index.ts]: now i'm being annoying, but can we name this "createCardTypeProfileV2"

**Review by ChriJ — APPROVED** [review]

File: model/InternalOrganisationConfig.ts (hunk 7 above)
Comment (kaffe2affe) [reply model/InternalOrganisationConfig.ts:20]: We'll add more later.

File: model/OrganisationConfig.ts (hunk 3 above)
Comment (kaffe2affe) [reply model/OrganisationConfig.ts]: Changed to use CardUsage.
Missing props will be added later.

**Pull request merged**